app/models/user.rb

```diff
@@ -96,6 +97,20 @@ def assign_barcode
     return self.barcode
   end
 
+  # Checks pin patterns against 
+  # the following examples:
+  # 1111, 2929, 0003, 5999. 
+  # Sierra will return the following
+  # error message if PIN is invalid:
+  # "PIN is not valid : PIN is trivial"
+  def validate_pin_pattern
+    if pin.scan(/(.)\1{2,}/).empty? && pin.scan(/(..)\1{1,}/).empty? == true
+      true
+    else
+      errors.add(:pin, 'format is invalid.')
```

Comment:
How about this message:  "Pin must be a 4-digit number with no repeating segments."

Reply:
Let me run it by Courtney.

Reply:
Talked to Courtney. We're going with "Your PIN does not meet our requirements. Please try again." .